portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,54 +14,86 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
 import java.util.ArrayList;
 import java.util.List;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private List<String> comments = new ArrayList<String>();
+    // Data Structure  
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     @Override 
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        Query query = new Query("Comment");
+
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Comment> comments = new ArrayList<>();
+
+
+        for (Entity entity : results.asIterable()) {
+            String name = (String) entity.getProperty("name");
+            String pageComment = (String )entity.getProperty("comment");
+            Comment comment = new Comment(name, pageComment);
+            comments.add(comment);
+        }
+
         Gson gson = new Gson();
-        String json = gson.toJson(comments);
         response.setContentType("application/json;");
-        response.getWriter().println((json));    
+        response.getWriter().println(gson.toJson(comments));    
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         // Get the input from the form.
         String comment = getComment(request);
 
-        // Add comment to comments Array.
-        if (comment != null) {
-            comments.add(comment);
+        // Get the name from the form.
+        String name = getName(request);
+        
+        // Create an entity representing the users comment. 
+        Entity commentEntity = new Entity("Comment");
+        
+        // Ensuring whether or not the user's comment and name werent null.
+        if (comment != null && name != null) {
```

Comment:
I like this check on null! But if the name and comment are null, should we still save the Entity to datastore?

Reply:
Acknowledged

Reply:
very tiny nit: Move the instantiation into this if statement would save you creating an empty object (but garbage collection would remove it anyway...)